### PR TITLE
Merge main into simplifier pass

### PR DESCRIPTION
**Pull request by gxyd**

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -1281,6 +1289,8 @@ jobs:
             cd integration_tests
             ./run_tests.py -b fortran -j1
             ./run_tests.py -b fortran -f -j1
+            ./run_tests.py -b fortran -j1 --experimental-simplifier
```
Comment (gxyd) [inline .github/workflows/CI.yml:1292]: @certik @Pranavchiku , just a gentle ping. This is **expected**.

These changes came directly after merge (as there were no merge conflicts for these changes), and slowly diff between *main* and *simplifier_pass* will vanish.

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,7 @@ endif ()
```
Comment (gxyd) [inline integration_tests/CMakeLists.txt:9]: @assem2002 , can you please give a quick look to see that the tests actually run with *simplifier_pass* turned on right?

Comment (gxyd) [reply integration_tests/CMakeLists.txt:9]: I didn't phrase the question correctly, but my point is simply, that if you can check that the changes after merging *main* here look alright to you?

**Comment by certik** [comment]
This is great. Yes, changes should slowly vanish, until only one change remains: making the experimental_simplifier=true, otherwise everything else is merged.

File: integration_tests/CMakeLists.txt (hunk 2 above)
Comment (assem2002) [reply integration_tests/CMakeLists.txt:9]: They exist at the same line number in main branch. 
https://github.com/lfortran/lfortran/blob/21630bc62c7fc0e98fdafe65873910f865bde5b1/integration_tests/CMakeLists.txt#L12C1-L14C1

Comment (assem2002) [reply integration_tests/CMakeLists.txt:9]: Yeah. This `--experimental_simplifier` option is useless on simplifier branch as has `compiler_option.po.experimental_simplifier` set to  TRUE by default.

File: src/libasr/pass/simplifier.cpp [hunk 3]
```diff
@@ -904,7 +904,7 @@ class ArgSimplifier: public ASR::CallReplacerOnExpressionsVisitor<ArgSimplifier>
 
     ArgSimplifier(Allocator& al_, ExprsWithTargetType& exprs_with_target_, bool realloc_lhs_) :
         al(al_), current_body(nullptr), exprs_with_target(exprs_with_target_),
-        realloc_lhs(realloc_lhs_) {}
+        realloc_lhs(realloc_lhs_) {(void)realloc_lhs; /*Silence-Warning*/}
```
Comment (assem2002) [inline src/libasr/pass/simplifier.cpp:907]: maybe you can undo that to remain as a diff between the two branch (maybe you can leave it, It causes no harm)

Comment (gxyd) [reply src/libasr/pass/simplifier.cpp:907]: I think this change is in *main* as well, it makes sense to me to have this here as well, reducing the diff size between *main* and *simplifier_pass* branches

**Pull request merged**